pika/connection.py

```diff
@@ -1067,7 +1127,75 @@ def __init__(self,
         if on_close_callback:
             self.add_on_close_callback(on_close_callback)
 
-        self.connect()
+        self._set_connection_state(self.CONNECTION_INIT)
+
+        if self._internal_connection_workflow:
+            # Kick off full-stack connection establishment. It will complete
+            # asynchronously.
+            self._adapter_connect_stack()
+        else:
+            # Externally-managed connection workflow will proceed asynchronously
+            # using adapter-specific mechanism
+            LOGGER.debug('Using external connection workflow.')
+
+    def _init_connection_state(self):
```

Comment:
Moved here to be close to constructor to make it easier to correlate member  variables.